src/hevm/src/EVM/SymExec.hs

```diff
@@ -364,8 +367,31 @@ simplify e = if (mapExpr go e == e)
       | a == (Lit 0) = v
       | otherwise = o
     go o@(And a (And b c))
+      | a == c = (And a b)
       | a == b = (And b c)
       | otherwise = o
+    go o@(Add a b)
+      | b == (Lit 0) = a
+      | a == (Lit 0) = b
+      | otherwise = o
+    go o@(EVM.Types.LT (Lit a) (Lit b))
+      | (a < b) = Lit 1
+      | otherwise = Lit 0
+    go o@(IsZero (CallValue x))
+      | x == 0 = Lit 0x1
+      | otherwise = Lit 0x0
+    -- we write at least 32, so if x <= 32, it's FALSE
+    go o@(EVM.Types.LT (BufLength (WriteWord {})) x)
+      | x <= Lit 0x32 = Lit 0x0
+      | otherwise = o
+    -- we write at least 32, so if x < 32, it's TRUE
+    go o@(EVM.Types.GT (BufLength (WriteWord {})) x)
+      | x < Lit 0x32 = Lit 0x1
```

Comment:
Shouldn't it be as follows? Or am I misunderstanding?

```suggestion
      | x < Lit 0x32 = Lit 0x0
```

Reply:
Let's do this step by step so I also understand a bit better :)

1. We write at least 32 bits. So we have ` o@(EVM.Types.GT y x)` where `y`>=32.
2. Simplify to `GT y x`, where `y`>=32.
3. Let's say we have `x` < 32, say 31 (largest `x` possible), and y=32 (smallest `y` possible). Then: `GT 32 31`.
4. Since 32>31, we have `true`, i.e. `0x01` as a result

So I believe this makes this rule correct? I am assuming that `GT a b` is operating as `a>b`.

Reply:
ah yes of course, I was confused. 

However, `0x32` is not the same as `32`, so the rule is still wrong but for a different reason :sweat_smile:

Reply:
Oh waaaaait!! OOooops. Good catch :100:

Reply:
OK, fixed to 32!